tests/__init__.py

```diff
@@ -66,16 +68,30 @@ def mkosi(
             "--qemu-mem=4G",
             verb,
             *args,
-        ], user=user, group=group)
+        ], stdin=stdin, stdout=sys.stdout, user=user, group=group)
 
     def build(self, options: Sequence[str] = (), args: Sequence[str] = ()) -> CompletedProcess:
-        return self.mkosi("build", [*options, "--debug", "--force"], args, user=INVOKING_USER.uid, group=INVOKING_USER.gid)
+        return self.mkosi(
+            "build",
+            [*options, "--debug", "--force"],
+            args,
+            stdin=sys.stdin,
+            user=INVOKING_USER.uid,
+            group=INVOKING_USER.gid,
+        )
 
     def boot(self, options: Sequence[str] = (), args: Sequence[str] = ()) -> CompletedProcess:
-        return self.mkosi("boot", [*options, "--debug"], args)
+        return self.mkosi("boot", [*options, "--debug"], args, stdin=sys.stdin)
 
     def qemu(self, options: Sequence[str] = (), args: Sequence[str] = ()) -> CompletedProcess:
-        return self.mkosi("qemu", [*options, "--debug"], args, user=INVOKING_USER.uid, group=INVOKING_USER.gid)
+        return self.mkosi(
+            "qemu",
+            [*options, "--debug"],
+            args,
+            stdin=sys.stdin,
```

Comment:
I'm not sure if this is the culprit, but when I try to use pytest, I get the following:
```
$ pytest -m integration -v tests/test_initrd.py::test_initrd
...
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
tests/__init__.py:76: in build
    return self.mkosi(
tests/__init__.py:58: in mkosi
    return run([
mkosi/run.py:207: in run
    return subprocess.run(
/usr/lib64/python3.12/subprocess.py:548: in run
    with Popen(*popenargs, **kwargs) as process:
/usr/lib64/python3.12/subprocess.py:992: in __init__
    errread, errwrite) = self._get_handles(stdin, stdout, stderr)
/usr/lib64/python3.12/subprocess.py:1708: in _get_handles
    p2cread = stdin.fileno()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <_pytest.capture.DontReadFromInput object at 0x7f162803ba40>

    def fileno(self) -> int:
>       raise UnsupportedOperation("redirected stdin is pseudofile, has no fileno()")
E       io.UnsupportedOperation: redirected stdin is pseudofile, has no fileno()

/usr/lib/python3.12/site-packages/_pytest/capture.py:220: UnsupportedOperation
```

With `-s`, it fails later.

Reply:
This should be fixed. I didn't notice because I always run with `-s`.

Can you get me a stacktrace of what fails later on?